docker/amdvlk.Dockerfile

```diff
@@ -62,34 +62,42 @@ RUN wget -P /usr/bin/ https://storage.googleapis.com/git-repo-downloads/repo \
     && cd /vulkandriver/drivers/spvgen/external \
     && python3 fetch_external_sources.py
 
-# Copy update script into container
+# Copy update and coverage report scripts into container.
 COPY docker/update-llpc.sh /vulkandriver/
+COPY docker/generate-coverage-report.sh /vulkandriver/
 
 # Build LLPC.
 WORKDIR /vulkandriver/builds/ci-build
-RUN EXTRA_FLAGS="" \
+RUN EXTRA_COMPILER_FLAGS=() \
+    && EXTRA_LINKER_FLAGS=() \
+    && EXTRA_FLAGS=() \
     && if echo "$FEATURES" | grep -q "+gcc" ; then \
-         EXTRA_FLAGS="$EXTRA_FLAGS -DCMAKE_C_COMPILER=gcc"; \
-         EXTRA_FLAGS="$EXTRA_FLAGS -DCMAKE_CXX_COMPILER=g++"; \
+         EXTRA_FLAGS+=("-DCMAKE_C_COMPILER=gcc"); \
+         EXTRA_FLAGS+=("-DCMAKE_CXX_COMPILER=g++"); \
        fi \
     && if echo "$FEATURES" | grep -q "+clang" ; then \
-         EXTRA_FLAGS="$EXTRA_FLAGS -DCMAKE_C_COMPILER=clang"; \
-         EXTRA_FLAGS="$EXTRA_FLAGS -DCMAKE_CXX_COMPILER=clang++"; \
-         EXTRA_FLAGS="$EXTRA_FLAGS -DLLVM_USE_LINKER=lld"; \
-         EXTRA_FLAGS="$EXTRA_FLAGS -DCMAKE_EXE_LINKER_FLAGS=-fuse-ld=lld"; \
-         EXTRA_FLAGS="$EXTRA_FLAGS -DCMAKE_SHARED_LINKER_FLAGS=-fuse-ld=lld"; \
+         EXTRA_FLAGS+=("-DCMAKE_C_COMPILER=clang"); \
+         EXTRA_FLAGS+=("-DCMAKE_CXX_COMPILER=clang++"); \
+         EXTRA_FLAGS+=("-DLLVM_USE_LINKER=lld"); \
+         EXTRA_LINKER_FLAGS+=("-fuse-ld=lld"); \
        fi \
     && if echo "$FEATURES" | grep -q "+shadercache" ; then \
-         EXTRA_FLAGS="$EXTRA_FLAGS -DLLPC_ENABLE_SHADER_CACHE=1"; \
+         EXTRA_FLAGS+=("-DLLPC_ENABLE_SHADER_CACHE=1"); \
+       fi \
+    && if echo "$FEATURES" | grep -q "+coverage" ; then \
+         EXTRA_COMPILER_FLAGS+=("-fprofile-instr-generate=/vulkandriver/profile%4m.profraw" "-fcoverage-mapping"); \
+         EXTRA_LINKER_FLAGS+=("-fprofile-instr-generate=/vulkandriver/profile%4m.profraw" "-fcoverage-mapping"); \
        fi \
     && if echo "$FEATURES" | grep -q "+sanitizers" ; then \
-         EXTRA_FLAGS="$EXTRA_FLAGS -DXGL_USE_SANITIZER=Address;Undefined"; \
-         echo "export ASAN_OPTIONS=detect_leaks=0" >> /vulkandriver/env.sh; \
-         echo "export LD_PRELOAD=$(clang -print-file-name=libclang_rt.asan-x86_64.so)" >> /vulkandriver/env.sh; \
+         EXTRA_FLAGS+=("-DXGL_USE_SANITIZER=Address;Undefined"); \
+         echo "export ASAN_OPTIONS=detect_leaks=0" >> env.sh; \
+         echo "export LD_PRELOAD=$(clang -print-file-name=libclang_rt.asan-x86_64.so)" >> env.sh; \
```

Comment:
I think this writes to `/vulkandriver/builds/ci-build/env.sh` instead of `/vulkandriver/env.sh`, which is still read below and in `llpc.Dockerfile`?

Reply:
Thanks, fixed.